File: LmsPlusPlus.Api/Infrastructure/Group.cs
namespace LmsPlusPlus.Api.Infrastructure;

public record Group
{
    public long Id { get; set; }
    public string Name { get; set; } = null!;
    public long TopicId { get; set; }
    public Topic Topic { get; set; } = null!;
    public ICollection<User> Users { get; set; } = null!;
}

File: LmsPlusPlus.Client/src/domain/Group.ts
import { User } from "./User"

export class Group {
    readonly id: number
    readonly name: string
    readonly users: readonly User[]
    readonly topicId: number

    constructor(id: number, name: string, users: readonly User[], topicId: number) {
        this.id = id
        this.name = name
        this.users = users
        this.topicId = topicId
    }
}
